silverback/middlewares.py

```diff
@@ -66,34 +64,40 @@ def fix_dict(data: dict, recurse_count: int = 0) -> dict:
         return message
 
     def _create_label(self, message: TaskiqMessage) -> str:
-        if message.task_name == "block":
-            args = f"[block={message.args[0].hash.hex()}]"
-
-        elif "event" in message.task_name:
-            args = f"[txn={message.args[0].transaction_hash},log_index={message.args[0].log_index}]"
+        if labels_str := (
+            ",".join(f"{k}={v}" for k, v in message.labels.items() if k != "task_name")
+        ):
+            return f"{message.task_name}[{labels_str}]"
 
         else:
-            args = ""
-
-        return f"{message.task_name}{args}"
+            return message.task_name
 
     def pre_execute(self, message: TaskiqMessage) -> TaskiqMessage:
-        if message.task_name == "block":
+        message.labels["task_name"] = message.task_name
```

Comment:
question: What's up with this?

Reply:
`task_name` is now the actual function name, not the constructed task name I gave it from before.

now, the thing to look at is `task_type` which groups all the tasks by buckets of how to work with the handler it has (events, blocks, startup/shutdown, etc.)

Reply:
I guess what I was getting at is why assign a prop to a label that's already on the object?  Seems redundant.